### PR TITLE
workload/schemachanger: correctly filter out unknown schema change errors

### PR DESCRIPTION
**Pull request by fqazi**
When code was added to handle unknown schema errors inside the schemachanger workload, the
regex used had a flaw that did not correctly quote the errors allowing no retries to occur. This meant
we never address the original errors but just avoided them due to low probability.

This PR also cleans up some debug code accidentally added in an older PR.

### CODE REVIEW

File: pkg/workload/schemachange/error_screening.go [hunk 1]
```diff
@@ -1116,7 +1116,7 @@ SELECT count(*) FROM %s
 }
 
 var (
-	regexpUnknownSchemaErr = regexp.MustCompile(`unknown schema \[\d+\]`)
+	regexpUnknownSchemaErr = regexp.MustCompile(`unknown schema "\[\d+]"`)
```
Comment (chengxiong-ruan) [inline pkg/workload/schemachange/error_screening.go:1119]: just want to confirm that we're trying to match something like `unknown schema "[1]"` ?

**Review by chengxiong-ruan — APPROVED** [review]
good find!

File: pkg/workload/schemachange/error_screening.go (hunk 1 above)
Comment (fqazi) [reply pkg/workload/schemachange/error_screening.go:1119]: Yes the errors have the form: unknown schema "[1]". I'll add a comment to make it clear

**Comment by fqazi** [comment]
@chengxiong-ruan TFTR!

bors r+

**Pull request merged**